tldextract/cache.py

```diff
@@ -113,7 +110,7 @@ def get(self, namespace: str, key: Union[str, Dict[str, Hashable]]) -> object:
             raise KeyError("namespace: " + namespace + " key: " + repr(key)) from None
 
     def set(
-        self, namespace: str, key: Union[str, Dict[str, Hashable]], value: object
+        self, namespace: str, key: str | dict[str, Hashable], value: object
```

Comment:
This syntax `|` requires [Python 3.10 or newer](https://peps.python.org/pep-0604)